sklearn/tests/test_metaestimators.py

```diff
@@ -151,3 +159,110 @@ def score(self, X, y, *args, **kwargs):
             assert not hasattr(delegator, method), (
                     "%s has method %r when its delegate does not"
                     % (delegator_data.name, method))
+
+
+def _generate_meta_estimator_instances_with_pipeline():
+    """Generate instances of meta-estimators fed with a pipeline
+
+    Are considered meta-estimators all estimators accepting one of "estimator",
+    "base_estimator" or "estimators".
+    """
+    for _, Estimator in sorted(all_estimators()):
+        sig = list(signature(Estimator).parameters)
+
+        if "estimator" in sig or "base_estimator" in sig:
+            if issubclass(Estimator, RegressorMixin):
+                estimator = make_pipeline(TfidfVectorizer(), Ridge())
+                param_grid = {"ridge__alpha": [0.1, 1.0]}
+            else:
+                estimator = make_pipeline(TfidfVectorizer(),
+                                          LogisticRegression())
+                param_grid = {"logisticregression__C": [0.1, 1.0]}
+
+            if "param_grid" in sig or "param_distributions" in sig:
+                # SearchCV estimators
+                yield Estimator(estimator, param_grid)
```

Comment:
This raises a UserWarning:

```
UserWarning: The total space of parameters 2 is smaller than n_iter=10.
Running 2 iterations. For exhaustive searches, use GridSearchCV.
```

for `RandomSearchCV`. I think this is okay to ignore with a `ignore_warnings(category=UserWarning)` decorator on `test_meta_estimators_delegate_data_validation`.

Reply:
Done, but I set n_iter=2 for RandomizedSearchCV instead of ignoring the warnings for the whole test.